api/src/bigquerytest/java/org/pmiops/workbench/api/CohortBuilderControllerTest.java

```diff
@@ -233,7 +235,38 @@ public void getCriteriaByTypeAndParentId_Phecodes() throws Exception {
                         .count(0L)
                         .conceptId(0L));
 
-        verify(mockCriteriaDao).findCriteriaByTypeAndParentId("PHECODE", 0L);
+        verify(mockCriteriaDao).findCriteriaByTypeAndParentIdOrderByCodeAsc("PHECODE", 0L);
+        verifyNoMoreInteractions(mockCriteriaDao);
+    }
+
+    @Test
+    public void getCriteriaTreeQuickSearch() throws Exception {
+        org.pmiops.workbench.cdr.model.Criteria expectedCriteria =
+                new org.pmiops.workbench.cdr.model.Criteria()
+                        .id(1L)
+                        .type("PHECODE")
+                        .name("Intestinal infection")
+                        .group(true)
+                        .selectable(true)
+                        .count("0")
+                        .conceptId("0");
+
+        when(mockCriteriaDao
```

Comment:
We should be able to have tests using a real criteria DAO instead of a mock... that would test that we set up our DAO logic appropriately... does that not work?

Reply:
I agree, but since we have switched to using mysql full text search, i'm not sure how to get H2 in memory database to handle the match() and against() functions within a unit test. I definitely would like a test case for this but am punting now until i can find a solution for this.

Reply:
Boo...

http://h2-database.66688.n3.nabble.com/MySQL-Compatibility-Mode-Fulltext-MATCH-function-td4033152.html

There is an FT_SEARCH thing in H2, but it requires its own setup and has completely different SQL. It sucks that this isn't handled in a more standard fashion...

There aren't a lot of great options here. Only things I can come up with:

1) Have this test run against MySQL in docker running locally, instead of H2.
2) Generate the DAO interface using some macro which uses one set of SQL for MySQL (when running in source normally) and a different set of SQL for H2 (when running in tests.)

I don't love either of these options. I don't love just having mocks here, either. 

If we are going to use a mock for testing interactions with this method, let's make sure we can still use an in-memory DAO for testing interactions with other methods.

Reply:
Absolutely. This is the only dao method that isn't tested against H2 currently. So if we have more of these cases pop up I think we can do something with option 1 above. I feel like option 2 only test sql that would never run in real mysql env, so seems pointless.